src/qibolab/couplers.py

```diff
@@ -0,0 +1,39 @@
+from dataclasses import dataclass, field
+from typing import Dict, Optional, Union
+
+from qibolab.channels import Channel
+from qibolab.qubits import Qubit, QubitId
+
+CouplerId = Union[str, int]
+"""Type for Coupler names."""
+
+
+@dataclass
+class Coupler:
+    """Representation of a physical coupler.
+
+    Coupler objects are instantiated by :class:`qibolab.platforms.platform.Platform`
+    but they are passed to instrument designs in order to play pulses.
+
+    Args:
+        name (int, str): Qubit number or name.
+
+        flux (:class:`qibolab.platforms.utils.Channel`): Channel used to
+            send flux pulses to the qubit.
+        Other characterization parameters for the coupler, loaded from the runcard.
```

Comment:
For dataclasses you can document fields under their definition (see next comment).